File: 11-container-with-most-water/most.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = 0
        length = len(height)
        head = 0
        tail = length -1 
        while tail - head > 0:
            potArea = min(height[head],height[tail]) * (tail-head) 
            area = max(area,potArea)
            if height[head] > height[tail]:
                tail=tail-1
            else:
                head=head+1
        
        return area
        

File: 11-container-with-most-water/most.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let area = 0;
    let length = height.length;
    let head = 0;
    let tail = length-1;
    while (tail - head > 0){
        let potArea = Math.min(height[head],height[tail]) * (tail - head);
        area = Math.max(area, potArea);
        if (height[head] > height[tail]){
            tail--;
        }else{
            head++;
        }
        
    }
    return area;
};